commit debceb4a7fc39af77a98365a4012e988c3a3aad7
Author: Jeremy Huddleston <jeremyhu@apple.com>
Date:   Tue Dec 20 21:26:24 2011 -0800

    Link with -undefined dynamic_lookup
    
    dock-support.o contains some symbols which may not be present at link
    time on older OS versions.  These symbols are annotated with weak_import
    and checked for at runtime, so it was possible to build on Snow Leopard
    and run on Leopard, but it was not possible to build on Leopard since
    these symbols are not available at link time.  This change makes it
    possible to build on Leopard and places the burden of resolving those
    symbols on dyld.
    
    Resolves: https://trac.macports.org/ticket/32609
    
    Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>

diff --git a/src/Makefile.am b/src/Makefile.am
index 1eeb8c6..30ea33b 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -28,7 +28,8 @@ AM_CFLAGS = $(QUARTZWM_CFLAGS) $(CWARNFLAGS)
 quartz_wm_LDFLAGS = $(QUARTZWM_LIBS) \
 	-framework AppKit \
 	-framework Foundation \
-	-framework ApplicationServices
+	-framework ApplicationServices \
+	-undefined dynamic_lookup
 
 quartz_wm_LDADD = dock-support.o
 
